server/app/views/admin/programs/ProgramBlockView.java

```diff
@@ -42,4 +51,48 @@ protected final DivTag renderProgramInfo(ProgramDefinition programDefinition) {
             editDetailsButton)
         .withClasses("bg-gray-100", "text-gray-800", "shadow-md", "p-8", "pt-4", "-mx-2");
   }
+
+  /** Renders a div presenting the predicate definition for the admin. */
+  protected final DivTag renderExistingPredicate(
+      String blockName,
+      PredicateDefinition predicateDefinition,
+      ImmutableList<QuestionDefinition> questionDefinitions) {
+    DivTag container = div();
+
+    if (predicateDefinition
+        .computePredicateFormat()
+        .equals(PredicateDefinition.PredicateFormat.SINGLE_QUESTION)) {
+      return container.with(
+          text(predicateDefinition.toDisplayString(blockName, questionDefinitions)));
+    } else if (!predicateDefinition
```

Comment:
The control flow is a little awkward here with partial processing and trying to ensure compute doesn't return an unexpected value.

Consider capturing compute to a variable then switch on it and have an explicit break for OR_OF_SINGLE_LAYER_ANDS that then flows to the next section of code, and a default that throws the exception.

Reply:
I wrote it with a switch the first time and didn't like how big the case ended up being for `OR_OF_SINGLE_LAYER_ANDS`, especially since there's two cases to cover within that case. Could have extracted it to a method but that didn't really help readability imo, just added more indirection.